### PR TITLE
Moved the start bot functionality

### LINKED ISSUE #160: Move start Bot functionality
As a end-user I want to execute bots from the overview.

---

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] The execution icon on the bot card in the overview should receive functionality 
- [ ] Block execution of bots that are not fully configured
- [ ] the id should no longer be hardcoded, but instead be retrieved from the bot container component

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #160 but builds up on Daniels Branch 95_proof_of_concept, therefore this should be merged first.

## Changes
I moved the functionality from the modeler sidebar to the robotContainer. I also created a small dummy function `isBotExecutable()` to implement the behavior when the bot is not ready to be executed. I would love to test this manually, but I did not understand exactly how to get the local client on my machine, maybe @WolfgangDaniel can help me there.

## Screenshots
grafik

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: client/src/utils/botExecution.jsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// is a dummy function
+const isBotExecutable = () => {
+    if (Math.random() > 0.49) {
+        return true;
+    }
+    return false;
+};
+export default isBotExecutable;
```
Comment (lukashueller) [inline client/src/utils/botExecution.jsx:8]: One small thing: In itself a great idea to include something like math.random, however it confused me a lot when testing it. With some clicks I got an alert window, with other clicks not. I thought the app was not responding. Maybe next time you might just return a constant alert with "... is not implemented yet"

**Review by lukashueller — COMMENTED** [review]
Please reconsider whether the Math.random should stay in, or explain to me what the purpose was.
Then you have my full agreement.

File: client/src/utils/botExecution.jsx (hunk 1 above)
Comment (WolfgangDaniel) [reply client/src/utils/botExecution.jsx:8]: Good point. I haven't thought of it myself but I agree with Lukas :)

**Review by WolfgangDaniel — APPROVED** [review]

File: client/src/utils/botExecution.jsx (hunk 1 above)
Comment (SanJSp) [reply client/src/utils/botExecution.jsx:8]: The isBotExecutable function is to simply a dummy function that is to be implemented in this sprint. As I would want the error handling functionality to be shown (is an acceptance criterion) I wanted it to be displayed. But I also wanted to show, that the current functionality is still there. Therefore, I do the error handling with a random probability to spice things up.

**Review by lukashueller — APPROVED** [review]

File: server/controllers/robotJobController.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+const ssotToRobotParser = require('../services/SsotToRobotParsing/SsotToRobotParser.js');
+const ssotModels = require('../models/singleSourceOfTruthModel.js');
+const robotJobs = [];
```
Comment (lukashueller) [inline server/controllers/robotJobController.js:4]: image

Please have a look at those ESLint Warnings. Then it should be fine from my side.

Comment (SanJSp) [reply server/controllers/robotJobController.js:4]: For those to go away #157 needs to be merged first

Comment (WolfgangDaniel) [reply server/controllers/robotJobController.js:4]: Yes, the first warning will go away with #157 and the second one will stay but we have this warning with every Controller

**Review by WolfgangDaniel — APPROVED** [review]

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
Please change the name of Ssotretrieval.js to ssotRetrieval.js
Otherwise it will throw an error because in the imports you are referring to ssotRetrieval

**Review by WolfgangDaniel — APPROVED** [review]

**Review by lukashueller — APPROVED** [review]

**Pull request merged**